storage/block_devices_ops.go

```diff
@@ -541,21 +545,593 @@ func partitionUsingParted(bd *BlockDevice, dryRun *[]string, wholeDisk bool) err
+func findMode(dryRun *DryRunType) string {
+	if dryRun == nil {
+		return "dryRun"
+	}
+
+	return "partition"
+}
+
+func checkRaidSpan(bd *BlockDevice, ForceDestructive bool, dryRun *DryRunType, scans *preScanResults) error {
+	log.Debug("checkRaidSpan: Running with ForceDestructive:%v in %s mode", ForceDestructive, findMode(dryRun))
+	if ForceDestructive && dryRun == nil {
+		return nil
+	}
+
+	RaidScanOutput := bytes.NewBuffer(nil)
+
+	args := []string{"mdadm", "--detail", bd.GetDeviceFile(), "--export"}
+
+	err := cmd.Run(RaidScanOutput, args...)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+
+	var AllRaidMembers = []string{}
+	for _, line := range strings.Split(RaidScanOutput.String(), "\n") {
+		fields := strings.Split(line, "=")
+		if len(fields) == 2 {
+			if strings.HasPrefix(fields[0], "MD_DEVICE") && strings.HasSuffix(fields[0], "DEV") {
+				AllRaidMembers = append(AllRaidMembers, fields[1])
+			}
+		}
+	}
+
+	sort.Strings(AllRaidMembers)
+	var raidMemberOtherDisks = []string{}
+	for _, raidmember := range AllRaidMembers {
+		found := false
+		for _, raidmemberblk := range scans.ScannedRaidList {
+			if raidmember == raidmemberblk.GetMappedDeviceFile() {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			raidMemberOtherDisks = append(raidMemberOtherDisks, raidmember)
+		}
+	}
+
+	if len(raidMemberOtherDisks) > 0 {
+		if dryRun != nil {
+			if len(*dryRun.OtherMediaResults) == 0 {
+				*dryRun.OtherMediaResults = append(*dryRun.OtherMediaResults, "Effect on Other Disks:")
+			}
+
+			*dryRun.OtherMediaResults = append(*dryRun.OtherMediaResults,
+				utils.Locale.Get("The Raid: [%s] has parts from other disks: [%s]",
+					bd.Name, strings.Join(raidMemberOtherDisks, ", ")))
+			setImpactOnOtherDisks(true)
+		} else {
+			log.Error("The Raid: [%s] has parts from other disks: [%s]",
+				bd.Name, strings.Join(raidMemberOtherDisks, ", "))
+			return errors.Errorf("Please check log and resolve the errors, before retrying installation")
+		}
+	}
+
+	return nil
+}
+
+func removeRaidType(bd *BlockDevice, ForceDestructive bool, disk string,
+	dryRun *DryRunType, scans *preScanResults) error {
+	if err := checkRaidSpan(bd, ForceDestructive, dryRun, scans); err != nil {
+		return err
+	}
+
+	raidParentFinder := func(b *BlockDevice) bool {
+		if b.FsType == "linux_raid_member" {
+			for _, ch := range b.Children {
+				if ch.Name == bd.Name {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	parent, err := reverseLookParent(bd, disk, raidParentFinder)
+	if dryRun == nil {
+		log.Debug("removeRaidType: Found parent: %s for child: %s", parent.GetDeviceFile(), bd.GetDeviceFile())
+	}
+
+	if parent == nil {
+		return err
+	}
+
+	if dryRun == nil {
+		log.Warning("Strategy 1: Failing the RAID part: %s failed for RAID: %s", parent.GetDeviceFile(), bd.Name)
+		args := []string{"mdadm", "--fail", bd.GetDeviceFile(), parent.GetDeviceFile()}
+
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			goto strategy2
```

Comment:
Hmmm....I wonder if instead of a goto we could have a named local function that could be re-used?

Reply:
Sure if that improves readability, I guess that can be done

Reply:
I don't see a lot of merit in moving to different function as it is a one off. This function is relatively simple, Is it okay if I leave like this I?

Reply:
My preference is that `goto`s aren't used in our code unless absolutely necessary.
If we had to break out of a deeply nested set of loops, or jump to a common error handling, etc. They should be for a critical exception, not the normal.

Reply:
cool. Done